### PR TITLE
Develop mel

### PR DESCRIPTION
**Pull request by meltemozcan**

### CODE REVIEW

**Comment by winniewytse** [comment]
@meltemozcan would you mind me directly making some changes to this pull request? For example, I have wanted to change the name of `get_ab()` to `beta_ab()` and change the inputs `d_est` to `delta` and `rho_est` to `rho`. Would you prefer me to make the changes in another branch or would it be okay for me to make these edits in your branch? Thanks!

**Comment by meltemozcan** [comment]
Hi @winniewytse! Yes, please feel free to make any edits you'd like! Your comment reminded me that I didn't get to share some ideas I had about possible function names, so I'll paste them here - please feel free to use/disregard as you'd like, we can also talk about it more during our next meeting. Thank you!

- al_2st() -> al_independent_t() or al_indp_t()
- al_crt2() -> al_lv2_crt() or al_l2_msrt()
- al_msrt2() -> al_lv2_msrt()
- apr_msrt2() -> alpr_lv2_msrt()
- get_ab() -> get_beta_shape()
- gamma_ab() -> get_gamma_shape()
- ep_crt2() -> ep_lv2_crt()
- ep_msrt2() -> ep_lv2_msrt()
- Jn_optimize() -> optimize_Jn() 
- n_2st() -> n_independendent_t() or n_indp_t()

> @meltemozcan would you mind me directly making some changes to this pull request? For example, I have wanted to change the name of `get_ab()` to `beta_ab()` and change the inputs `d_est` to `delta` and `rho_est` to `rho`. Would you prefer me to make the changes in another branch or would it be okay for me to make these edits in your branch? Thanks!

**Comment by winniewytse** [comment]
Hey @meltemozcan! Thanks for sharing your suggestions for the function names! I like `al_indp_t()`, `optimize_Jn()`, and `n_indp_t()`, and may prefer keeping the name short for others. I'm still debating between clarity and simplicity for the function names. We can discuss more in our meeting. I think `get_beta_shape()` is good but the parameters for a Gamma distribution are called shape and rate/scale instead, so `get_gamma_shape()` may not be the most accurate. Or we can do `get_beta_par()` or `beta_par()` (similarly, `get_gamma_par()` or `gamma_par()`)? 

Another quick thought: would you like `Jn_crt2()` or `design_crt2()` better? 

And thanks for letting me make changes to this pull request! Just a list of items I will clean up:
- [x] Clone the` develop_mel` branch to a new branch `reparameterize` and remove the codes for reparameterization in the current branch
- [x] Change `get_ab()` to `beta_ab()` (for now)
- [x] Change the names of some input parameters for preciseness (`d_est` -> `delta`, etc.)
- [x] Clean up draws-related functions and precision-related functions
- [x] Tidy up unit tests

Besides, I found out that `do.call()` works nicely with a list of input parameters. I built on top of your codes and now with `params`, which is a list of arguments, we can do `do.call(criteria, append(J = J, n = n, params))` to define the loss function. This is just one line of code after setting up `params` in the way you did. I will push an update later. 

Winnie

**Comment by meltemozcan** [comment]
Good morning @winniewytse! Thank you for all your work going through the pull request and updating the parameter names in all the files! I like the `d_est `-> `delta`  and `d_sd` -> `delta_sd` a lot, it seems much clearer. 

> Hey @meltemozcan! Thanks for sharing your suggestions for the function names! I like `al_indp_t()`, `optimize_Jn()`, and `n_indp_t()`, and may prefer keeping the name short for others. I'm still debating between clarity and simplicity for the function names. We can discuss more in our meeting. I think `get_beta_shape()` is good but the parameters for a Gamma distribution are called shape and rate/scale instead, so `get_gamma_shape()` may not be the most accurate. Or we can do `get_beta_par()` or `beta_par()` (similarly, `get_gamma_par()` or `gamma_par()`)?

Yes, you're right that one parameter of gamma is rate/scale. I like the beta_par or get_beta_par format - one additional form we could consider is `beta_params()` or `get_beta_params()`, which again is a tradeoff between simplicity and clarity. Looking forward to discussing further in our next meeting!

> Another quick thought: would you like `Jn_crt2()` or `design_crt2()` better?

I think both are great! Although, do you think `design_crt2()` may be a bit too general? `get_design_crt2()` is a potential alternative, though it is a little longer. Just a very minor thought!
> . . .
> 
> Besides, I found out that `do.call()` works nicely with a list of input parameters. I built on top of your codes and now with `params`, which is a list of arguments, we can do `do.call(criteria, append(J = J, n = n, params))` to define the loss function. This is just one line of code after setting up `params` in the way you did. I will push an update later.
>
I love this! The function call looks so neat and clean after incorporating `do.call()`, especially since it seems to handle the list containing both character and numeric types! Thank you again for all your updates.

File: R/Jn_crt2.R [hunk 1]
```diff
@@ -76,55 +76,74 @@ Jn_crt2 <- function(d_est, d_sd, rho_est, rho_sd, rsq2 = 0, J = NULL,
...
+    min_n <- 1
 
-    n <- try(stats::uniroot(loss, interval = c(min_j, 1e8))$root, silent = TRUE)
+    n <- try(stats::uniroot(loss, interval = c(min_n, 1e8))$root, silent = TRUE)
     if (class(n) == "try-error") {
-      loss <- define_loss(solve_for_J = FALSE, squared = TRUE, ep, al,
-                          J = j_temp, n = n_temp,test = test,
-                          reparameterize = reparameterize, list_params = params)
+      loss <- function(n) {
+        (do.call(criteria, append(list(J = J, n = n), params)) - target)^2
```
Comment (meltemozcan) [inline R/Jn_crt2.R:121]: @winniewytse It looks like we can delete the define_loss() helper function now, nice!

**Pull request merged**